pkg/apis/apps/validation/validation_test.go

```diff
@@ -2263,6 +2289,28 @@ func TestValidateDaemonSetUpdate(t *testing.T) {
 		if errs := ValidateDaemonSetUpdate(&successCase.update, &successCase.old, corevalidation.PodValidationOptions{}); len(errs) != 0 {
 			t.Errorf("%q expected no error, but got: %v", testName, errs)
 		}
+		// Verification gate
```

Comment:
This whole block is convoluted - I don't get why it calls ValidateDaemonSetUpdate() twice.

Can you just either add a few simple cases with optional gate config, like the core tests (which was very nice!)?

Reply:
I may not write it well. I will improve it immediately